test/cql-pytest/test_permissions.py

```diff
@@ -39,3 +40,46 @@ def test_grant_applicable_data_and_role_permissions(cql, test_keyspace, cassandr
             for permission in all_permissions.difference(set(permissions)):
                 with pytest.raises((InvalidRequest, SyntaxException), match="support.*permissions"):
                     cql.execute(f"GRANT {permission} ON {resource} TO {user}")
+
+
+def eventually_authorized(fun, retries=100):
```

Comment:
Nitpick (feel free to reject): I personally prefer to see a timeout in seconds, because otherwise there is always down when you see "retries=100" what it means, and you need to go calculating (in this case it's easy, but when the delay is exponential, it's not).